### PR TITLE
feat: onboarding audience

### PR DESCRIPTION
**Pull request by lucianak**
New members should be added to the onboarding audience when they get a membership for the first time, and removed after they complete the onboarding journey

### CODE REVIEW

File: packages/backend-modules/mail/MailchimpInterface.js [hunk 1]
```diff
@@ -14,16 +14,24 @@ const MINIMUM_HTTP_RESPONSE_STATUS_ERROR = 400
...
+    },
+    buildAudienceApiUrl(email, audienceId) {
+      const hash = crypto
+        .createHash('md5')
+        .update(email.toLowerCase())
+        .digest('hex')
+
+      return this.buildApiUrl(audienceId, `/members/${hash}`)
     },
     buildMembersApiUrl(email) {
```
Comment (annatraussnig) [inline packages/backend-modules/mail/MailchimpInterface.js:28]:
```suggestion
    buildMembersApiUrl(email, audienceId = MAILCHIMP_MAIN_LIST_ID) {
```
if audienceId is undefined the function would use the given default, i.e. MAILCHIMP_MAIN_LIST_ID

Comment (annatraussnig) [reply packages/backend-modules/mail/MailchimpInterface.js:28]: then you could use `buildMembersApiUrl` everywhere and delete `buildAudienceApiUrl`

File: packages/backend-modules/mail/MailchimpInterface.js [hunk 2]
```diff
@@ -90,6 +98,50 @@ const MailchimpInterface = ({ logger }) => {
         throw new NewsletterMemberMailError({ error, email })
       }
     },
+    async updateMemberInAudience(email, data, audienceId) {
```
Comment (annatraussnig) [inline packages/backend-modules/mail/MailchimpInterface.js:101]: maybe here we could reuse `async updateMember(email, data)` but with a default audienceId as well, i.e. `async updateMember(email, data, audienceId = MAILCHIMP_MAIN_LIST_ID)` (if the rest of the logic is exactly the same)

File: packages/backend-modules/mail/MailchimpInterface.js [hunk 3]
```diff
@@ -14,16 +14,24 @@ const MINIMUM_HTTP_RESPONSE_STATUS_ERROR = 400
 const MailchimpInterface = ({ logger }) => {
   checkEnv(['MAILCHIMP_API_KEY', 'MAILCHIMP_URL', 'MAILCHIMP_MAIN_LIST_ID'])
   return {
-    buildApiUrl(path) {
-      return `${MAILCHIMP_URL}/3.0/lists/${MAILCHIMP_MAIN_LIST_ID}${path}`
+    buildApiUrl(audienceId, path) {
```
Comment (annatraussnig) [inline packages/backend-modules/mail/MailchimpInterface.js:17]:
```suggestion
    buildApiUrl(path, audienceId = MAILCHIMP_MAIN_LIST_ID) {
```
then it's safer if the function is used somewhere random we don't know about…

File: packages/backend-modules/mail/MailchimpInterface.js [hunk 4]
```diff
@@ -113,6 +165,21 @@ const MailchimpInterface = ({ logger }) => {
         throw new NewsletterMemberMailError({ error, email })
       }
     },
+    async archiveMemberInAudience(email, audienceId) {
```
Comment (annatraussnig) [inline packages/backend-modules/mail/MailchimpInterface.js:168]:
```suggestion
    async archiveMember(email, audienceId = MAILCHIMP_MAIN_LIST_ID) {
```

**Review by annatraussnig — COMMENTED** [review]
Looks pretty good :) 
Added a few JS suggestions left and right… Looking forward to seeing it in action!

File: packages/backend-modules/republik-crowdfundings/lib/Mail.js [hunk 5]
```diff
@@ -28,32 +28,46 @@ const {
...
-    name: 'ACCOMPLICE',
-    interestId: MAILCHIMP_INTEREST_NEWSLETTER_ACCOMPLICE,
-    visibleToRoles: ['accomplice'],
-  },
-  {
-    name: 'PROJECTR',
-    interestId: MAILCHIMP_INTEREST_NEWSLETTER_PROJECTR,
-  },
-])
+const mail = createMail(
```
Comment (lucianak) [inline packages/backend-modules/republik-crowdfundings/lib/Mail.js:36]: audience configuration isn't really used anywhere

File: packages/backend-modules/republik-crowdfundings/lib/Mail.js [hunk 6]
```diff
@@ -117,9 +131,42 @@ const getInterestsForUser = async ({
...
+
+  // TODO tbd, maybe merge this with NewsletterSubscription
+  const result = {
+    user,
+    status: MailchimpInterface.MemberStatus.Subscribed,
+  }
+  debug(result)
+  return result
+}
+
```
Comment (lucianak) [inline packages/backend-modules/republik-crowdfundings/lib/Mail.js:165]:
```suggestion
mail.addUserToAudience = addUserToAudience
```

File: packages/backend-modules/republik-crowdfundings/lib/Mail.js [hunk 7]
```diff
@@ -117,9 +131,42 @@ const getInterestsForUser = async ({
...
+  debug(data)
+
+  const mailchimp = MailchimpInterface({ console })
+  await mailchimp.updateMemberInAudience(email, data, audienceId)
+
+  // TODO tbd, maybe merge this with NewsletterSubscription
+  const result = {
+    user,
+    status: MailchimpInterface.MemberStatus.Subscribed,
```
Comment (lucianak) [inline packages/backend-modules/republik-crowdfundings/lib/Mail.js:160]: always gives back status `Subscribed` regardless of actual result

File: packages/backend-modules/mail/MailchimpInterface.js
Comment (lucianak) [inline packages/backend-modules/mail/MailchimpInterface.js:1]: @annatraussnig what do you think?

**Review by fuenkchen — CHANGES_REQUESTED** [review]
Not finished yet, but I thought, I send these comments already now.

File: packages/backend-modules/republik-crowdfundings/lib/Mail.js [hunk 8]
```diff
@@ -28,6 +28,8 @@ const {
   MAILCHIMP_INTEREST_NEWSLETTER_PROJECTR,
   MAILCHIMP_INTEREST_NEWSLETTER_ACCOMPLICE,
   MAILCHIMP_INTEREST_NEWSLETTER_CLIMATE,
+  MAILCHIMP_MAIN_LIST_ID,
```
[the remark this answers is not in the document]
Comment (lucianak) [reply packages/backend-modules/republik-crowdfundings/lib/Mail.js:31]: I think AUDIENCE is more explicit, but I didn't want to change too many things of the existing code just yet

File: packages/backend-modules/republik-crowdfundings/lib/activateMembership.js [hunk 9]
```diff
@@ -59,5 +59,5 @@ module.exports = async (membership, user, t, pgdb) => {
     )
   }
 
-  return updatedMembership
+  return { updatedMembership, hasActiveMembership }
```
[the remark this answers is not in the document]
Comment (fuenkchen) [reply packages/backend-modules/republik-crowdfundings/lib/activateMembership.js:62]: This one was resolved, but changing the return value here has an impact on the mutation `activateMembership` and on mutation `claimMembership` as far as I can see.

**Review by fuenkchen — APPROVED** [review]
lgtm

**Pull request merged**